### PR TITLE
Update rector config to resolve bootstrap issues with recent rector releases

### PR DESCRIPTION
**Pull request by mglaman**
## Description
Expands on #219

Fixes bootstrap file for PHPUnit with changes from Rector

## To Test
- Add steps to test this feature

## Drupal.org issue
[drupal.org/project/rector/issues/3346620](https://www.drupal.org/project/rector/issues/3346620)

### CODE REVIEW

**Comment by mglaman** [comment]
```
"Rector\Core\Exception\ShouldNotHappenException" thrown in             
         "/home/runner/drupal/vendor/palantirnet/drupal-rector/config/drupal-php
         unit-bootstrap-file.php" on line 14 while loading bootstrap file       
         /home/runner/drupal/vendor/palantirnet/drupal-rector/config/drupal-9/..
         /drupal-phpunit-bootstrap-file.php: The Drupal PHPUnit Bootstrap file  
         could not access the BootstrapFilesIncluder
```

`\Rector\Core\Autoloading\BootstrapFilesIncluder::includeBootstrapFiles` used to pass in `BootstrapFilesIncluder`. This doesn't. We need to fix the assumptions in this file.

**Comment by mglaman** [comment]
Change in `\Rector\Core\Autoloading\BootstrapFilesIncluder::tryRequireFile`

```
            (static function (string $bootstrapFile) use ($container): void {
                require_once $bootstrapFile;
            })($bootstrapFile);
```

It now passes in `\PHPStan\DependencyInjection\Container` as a global value

**Comment by mglaman** [comment]
Problem stated on https://github.com/rectorphp/rector/issues/7795

File: config/drupal-phpunit-bootstrap-file.php [hunk 1]
```diff
@@ -10,24 +10,8 @@
...
-
-/** @phpstan-ignore-next-line */
-$autoloadPaths = $parameterProvider->provideArrayParameter(\Rector\Core\Configuration\Option::AUTOLOAD_PATHS);
-if (count($autoloadPaths) === 0) {
-    throw new \RuntimeException('No autoload paths were specified.');
-}
-
 $drupalFinder = new DrupalFinder();
-$drupalFinder->locateRoot($autoloadPaths[0]);
+$drupalFinder->locateRoot(__DIR__);
```
Comment (mglaman) [inline config/drupal-phpunit-bootstrap-file.php:14]: This could break someone's setup if they had something _really goofy_ and custom going on.

**Review by mglaman — COMMENTED** [review]
One note

**Comment by goba** [comment]
Beautiful :) Thanks a lot!

File: config/drupal-phpunit-bootstrap-file.php (hunk 1 above)
Comment (agentrickard) [reply config/drupal-phpunit-bootstrap-file.php:14]: This is probably worth a docs note

**Review by agentrickard — COMMENTED** [review]
How do we document the `__DIR__` change in the PHP Unit bootstrap? 

I know this should "just work" but is there advice we can give at least?

**Comment by mglaman** [comment]
There's no possible fix beyond "open an issue and describe your setup"

We can no longer infer the location of Drupal to fix autoloading of test namespaces.

**Pull request merged**